rest-service/manager_rest/resource_manager.py

```diff
@@ -78,9 +79,32 @@ def update_execution_status(self, execution_id, status, error):
                 .format(execution.status, status))
         execution.status = status
         execution.error = error
+
         if status in ExecutionState.END_STATES:
             execution.ended_at = utils.get_formatted_timestamp()
-        return self.sm.update(execution)
+
+        res = self.sm.update(execution)
+        if status in ExecutionState.END_STATES:
+            self.start_queued_executions()
+        return res
+
+    def start_queued_executions(self):
+        queued_executions = self._get_queued_executions()
+
+        for e in queued_executions:
+            self.execute_queued_workflow(e)
+            if e.is_system_workflow:  # To avoid starvation of system workflows
+                break
+
+    def _get_queued_executions(self):
+        filters = {'status': ExecutionState.QUEUED_STATE}
+        queued_executions = [
+            e for e in self.list_executions(is_include_system_workflows=True,
```

Comment:
do we really need a list here? maybe we can just `return self.list_executions().items`..